### PR TITLE
Add new rule `require-media-caption`

### LINKED ISSUE #2496: Proposal for new a11y check to enforce that media elements have captions
Currently, Ember Template Lint does not enforce that media elements, such as `<audio>` and `<video>`, provide the `track` element with a `kind` attribute value of `captions` (`<track kind="captions" />`). Captions are essential for users who are deaf, as well as those for whom the media content is unavailable, to access the portion of the content conveyed via the audio track. 

This proposes a new rule to enforce that media elements provide a track with  captions.

Reference: - [Understanding Success Criterion 1.2.2: Captions (Prerecorded)](https://www.w3.org/WAI/WCAG21/Understanding/captions-prerecorded.html)

### PR DESCRIPTION
**Pull request by judithhinlung**
This PR adds the new rule require-media-caption which fixes #2496.

### background
Captions are essential for users who are deaf, as well as those for whom the media content is unavailable, to access synchronized media content. By providing a transcript of the spoken content as well as descriptions of other relevant non-speech information, captions allow users access to the information conveyed via the audio track.

Reference: - [Understanding Success Criterion 1.2.2: Captions (Prerecorded)](https://www.w3.org/WAI/WCAG21/Understanding/captions-prerecorded.html)

**Allowed**:
```
# examples
<audio><track kind="captions" /></audio>
<video muted="true"></video>
<video muted></video>
```
**Forbidden**:
```
# examples
<audio></audio>
<video><track /></video>
<video><track kind="descriptions" /></video>
<audio muted="false"></audio>
```

### Testing
Test Suites: 137 passed, 137 total
Tests:       5 skipped, 7534 passed, 7539 total
Snapshots:   985 passed, 985 total
Time:        109.104 s
Ran all test suites.
✨  Done in 115.11s.

### CODE REVIEW

File: test/unit/rules/require-media-caption-test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      template: '<video><track kind="subtitles" /></video>',
+      result: {
+        message: 'Media elements such as <audio> and <video> must have a <track> for captions.',
+        line: 1,
+        column: 0,
+        source: '<video><track kind="subtitles" /></video>',
+      },
+    },
+    {
+      template: '<audio muted="false"></audio>',
```
Comment (bmish) [inline test/unit/rules/require-media-caption-test.js:45]: Shouldn't we test with `muted=false` too?

File: test/unit/rules/require-media-caption-test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    '<audio muted="true"></audio>',
+    '<video muted></video>',
+    '<audio muted={{this.muted}}></audio>',
+    '<video><track kind="captions" /><track kind="descriptions" /></video>',
+  ],
+
+  bad: [
+    {
+      template: '<video></video>',
+      result: {
```
Comment (bmish) [inline test/unit/rules/require-media-caption-test.js:19]: Let's use `verifyResults` snapshot testing for all these test cases.

**Review by bmish — CHANGES_REQUESTED** [review]
Do you think this rule is suitable to become a `recommended` rule in the next major release?

**Comment by judithhinlung** [comment]
Thanks for the review @bmish , I have updated the tests to use verifyResults and added a test case validating muted=false fails as expected.

**Comment by judithhinlung** [comment]
> Do you think this rule is suitable to become a `recommended` rule in the next major release?

Yes I agree that this would be a good `recommended` rule.

File: docs/rule/require-media-caption.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+```hbs
+<video muted="true"></video>
+```
+
+```hbs
+<video muted></video>
+```
+
```
Comment (bmish) [inline docs/rule/require-media-caption.md:36]: I think we can remove this example. We don't really need to show different syntaxes for the same thing.
```suggestion
```

File: docs/rule/require-media-caption.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+```hbs
+<video><track /></video>
+```
+
+```hbs
+<video><track kind="descriptions" /></video>
+```
+
+This rule **allows** the following:
```
Comment (bmish) [inline docs/rule/require-media-caption.md:23]: Can we also show a `video` properly using captions?

**Review by bmish — APPROVED** [review]
Thanks!

**Pull request merged**